### PR TITLE
[NavigationDrawer] Fix over scrolled bottom navigation incorrectly showing the scrim and presenting view

### PR DESCRIPTION
**Pull request by randallli**
Fixes the over-scroll of the bottom navigation drawer so that it does not show the scrim and presenting view controller when over-scrolling the content.

We do this by setting the color of the scrim to the background color of the tracking scroll view.

The default scrim color is now exposed.

Before:
Simulator Screen Shot - iPhone 11 Pro - 2019-10-21 at 16 36 06

after:
Simulator Screen Shot - iPhone 11 Pro - 2019-10-21 at 16 33 16
Simulator Screen Shot - iPhone 11 Pro - 2019-10-21 at 16 33 11

This work started with @andrewoverton's https://github.com/material-components/material-components-ios/pull/8612

### CODE REVIEW

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h [hunk 1]
```diff
@@ -24,6 +24,21 @@
  Delegate for MDCBottomDrawerContainerViewController.
  */
 @protocol MDCBottomDrawerContainerViewControllerDelegate <NSObject>
+
+/**
+This method is called when the bottom drawer updates its value for scrollViewIsScrolledToBottom.
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h]: we should probably indicate that this is only activated when trackingScrollView is set

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h [hunk 2]
```diff
@@ -24,6 +24,21 @@
...
+
+@param containerViewController the container view controller of the bottom drawer.
+@param scrollViewIsScrolledToEndOfContent whether or not the scroll view is scrolled to the end of
+the content.
+*/
+- (void)
+    bottomDrawerContainerViewControllerDidUpdateValueForScrollViewIsScrolledToEndOfContent:
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h]: would we be able to shorten the naming here? it's a bit harder to comprehend. maybe `bottomDrawerContainerViewControllerDidReachEndOfContent`?

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 3]
```diff
@@ -377,4 +377,14 @@ - (void)bottomDrawerContainerViewControllerDidChangeYOffset:
   }
 }
 
+- (void)
+    bottomDrawerContainerViewControllerDidUpdateValueForScrollViewIsScrolledToEndOfContent:
+        (MDCBottomDrawerContainerViewController *)containerViewController
+                                                        scrollViewIsScrolledToEndOfContent:
+                                                            (BOOL)
+                                                                scrollViewIsScrolledToEndOfContent {
+  UIColor *normalScrimColor = self.scrimColor ?: [UIColor colorWithWhite:0 alpha:(CGFloat)0.32];
```
Comment (yarneo) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m]: may be worth commenting here on why we added this behavior and what its trying to solve. so future contributors will better understand.

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h (hunk 1 above)
Comment (andrewoverton) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h]: I renamed `scrollViewIsScrolledToBottom ` to `scrollViewIsScrolledToEndOfContent` so people wouldn't confuse it with the preexisting (and questionably named) `scrollViewIsDraggedToBottom` property, but failed to change it in some of the documentation, including this line.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 4]
```diff
@@ -377,4 +377,17 @@ - (void)bottomDrawerContainerViewControllerDidChangeYOffset:
   }
 }
 
+- (void)bottomDrawerContainerViewControllerDidReachEndOfContent:
+            (MDCBottomDrawerContainerViewController *)containerViewController
+                             scrollViewIsScrolledToEndOfContent:
+                                 (BOOL)scrollViewIsScrolledToEndOfContent {
+  if (self.trackingScrollView) {
+    // This logic is to mittigate b/119714330. Dragging the drawer further up when already at the
```
Comment (andrewoverton) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m]: mitigate is spelled wrong

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h [hunk 5]
```diff
@@ -24,6 +24,20 @@
  Delegate for MDCBottomDrawerContainerViewController.
  */
 @protocol MDCBottomDrawerContainerViewControllerDelegate <NSObject>
+
+/**
+This method is called when the bottom drawer updates its value for scrollViewIsScrolledToBottom when
```
Comment (andrewoverton) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.h]: I already left this comment but it became outdated--I renamed `scrollViewIsScrolledToBottom` to `scrollViewIsScrolledToEndOfContent` so people wouldn't confuse it with the preexisting (and questionably named) `scrollViewIsDraggedToBottom` property, but failed to change it in some of the documentation, including this line. So `scrollViewIsScrolledToBottom` should be changed to `scrollViewIsScrolledToEndOfContent`.

**Review by andrewoverton — APPROVED** [review]

**Pull request merged**